# <b> EDA on 2022 annual CDC survey data of 400k+ adults related to their health status.

### **Contributers:** 

- Ali Bin Kashif : [Know about Ali here.](https://www.linkedin.com/in/ali-bin-kashif/)
- Muhammad Aman : [Know about Aman here.](https://www.linkedin.com/in/muhammad-aman-5a9317288/)

### <b>About Dataset:</b>

According to the CDC, heart disease is a leading cause of death for people of most races in the U.S. (African Americans, American Indians and Alaska Natives, and whites). About half of all Americans (47%) have at least 1 of 3 major risk factors for heart disease: high blood pressure, high cholesterol, and smoking. Other key indicators include diabetes status, obesity (high BMI), not getting enough physical activity, or drinking too much alcohol. Identifying and preventing the factors that have the greatest impact on heart disease is very important in healthcare. In turn, developments in computing allow the application of machine learning methods to detect "patterns" in the data that can predict a patient's condition.

The dataset originally comes from the CDC and is a major part of the Behavioral Risk Factor Surveillance System (BRFSS), which conducts annual telephone surveys to collect data on the health status of U.S. residents. As described by the CDC: "Established in 1984 with 15 states, BRFSS now collects data in all 50 states, the District of Columbia, and three U.S. territories. BRFSS completes more than 400,000 adult interviews each year, making it the largest continuously conducted health survey system in the world.

### **Variables**:

There are 40 variables(columns) in this dataset.

- **State :** The U.S. state where the individual resides.

- **Sex :** Gender of the individual (Male or Female).

- **GeneralHealth :** Self-reported general health status of the individual.

- **PhysicalHealthDays :** Number of days in the past 30 days that physical health was not good.

- **MentalHealthDays :** Number of days in the past 30 days that mental health was not good.

- **LastCheckupTime :** Time since the last routine checkup or health examination.

- **PhysicalActivities :** Frequency of engaging in physical activities or exercises.

- **SleepHours :** Average number of hours of sleep per night.

- **RemovedTeeth :** Number of permanent teeth removed due to dental issues.

- **HadHeartAttack :** Whether the individual has had a heart attack.

- **HadAngina :** Whether the individual has experienced angina (chest pain or discomfort).

- **HadStroke :** Whether the individual has had a stroke.

- **HadAsthma :** Whether the individual has had asthma.

- **HadSkinCancer :** Whether the individual has had skin cancer.

- **HadCOPD :** Whether the individual has had Chronic Obstructive Pulmonary Disease (COPD).

- **HadDepressiveDisorder :** Whether the individual has had a depressive disorder.

- **HadKidneyDisease :** Whether the individual has had kidney disease.

- **HadArthritis :** Whether the individual has had arthritis.

- **HadDiabetes :** Whether the individual has had diabetes.

- **DeafOrHardOfHearing :** Whether the individual is deaf or hard of hearing.

- **BlindOrVisionDifficulty :** Whether the individual has blindness or vision difficulty.

- **DifficultyConcentrating :** Self-reported difficulty in concentrating.

- **DifficultyWalking :** Self-reported difficulty in walking.

- **DifficultyDressingBathing :** Self-reported difficulty in dressing or bathing.

- **DifficultyErrands :** Self-reported difficulty in running errands.

- **SmokerStatus :** Current smoking status of the individual (smoker, former smoker, non-smoker).

- **ECigaretteUsage :** Whether the individual uses e-cigarettes.

- **ChestScan :** Whether the individual has had a chest scan.

- **RaceEthnicityCategory :** Categorized race or ethnicity of the individual.

- **AgeCategory :** Categorized age group of the individual.

- **HeightInMeters :** Height of the individual in meters.

- **WeightInKilograms :** Weight of the individual in kilograms.

- **BMI :** Body Mass Index calculated from height and weight.

- **AlcoholDrinkers :** Whether the individual consumes alcohol.

- **HIVTesting :** Whether the individual has undergone HIV testing.

- **FluVaxLast12 :** Whether the individual received a flu vaccine in the last 12 months.

- **PneumoVaxEver :** Whether the individual has ever received a pneumonia vaccine.

- **TetanusLast10Tdap :** Time since the last tetanus vaccination (in the last 10 years, received Tdap).

- **HighRiskLastYear :** Whether the individual has been considered at high risk for the past year.

- **CovidPos :** Whether the individual tested positive for COVID-19.


# Problem Statement:

### **Understanding and Mitigating Cardiovascular Health Disparities:** An In-Depth Analysis of Risk Factors and Health Behaviors in the Adult Population.

The problem centers on identifying and understanding the factors that exert the most significant influence on heart disease prevalence.This exploration is crucial for healthcare initiatives and interventions aimed at prevention and management. The dataset encompasses a range of variables, including demographic information, health behaviors, chronic conditions, and COVID-19 status, providing a rich source for uncovering patterns, correlations, and disparities in cardiovascular health.

We're looking at big questions like what things make heart problems more likely, how mental and physical health are linked, what habits and steps help prevent heart issues, how other health problems affect the heart, and if issues like hearing or vision problems are connected to heart disease. We're also checking how smoking and e-cigarettes are used and how COVID-19 affects heart health.

Our main goal is to extract actionable insights from this wealth of data. We want to make plans that focus on specific things to help reduce differences in heart health and make everyone's heart healthier in the adult population.

# Research and Analytical Questions:

These research questions are given by ChatGPT according to our problem statement.

**Demographic Variation:**

- How does the prevalence of heart disease and major risk factors vary across different demographic groups (sex, age, race/ethnicity)?
Are there specific subgroups experiencing higher cardiovascular health disparities?
Risk Factor Patterns:

- What are the common patterns and combinations of major risk factors (high blood pressure, high cholesterol, smoking) within the dataset?
How do these patterns relate to the overall cardiovascular health of individuals?

**Impact of Chronic Conditions:**

- Among individuals with chronic conditions (diabetes, asthma, COPD), how does the presence of these conditions correlate with heart disease prevalence?
- Are certain chronic conditions more strongly associated with cardiovascular health disparities?

**Behavioral Correlations:**

- Explore the relationships between health behaviors (physical activity, sleep, alcohol consumption) and the prevalence of heart disease.
- Identify key behavioral factors that may contribute to cardiovascular health disparities.

**Effect of Mental Health on Heart Disease:**

- Investigate the interplay between mental health (mental health days, depressive disorders) and heart disease.
- Are individuals with certain mental health conditions more susceptible to cardiovascular issues?

**Sensory Impairments and Cardiovascular Health:**

- Analyze the impact of sensory impairments (deafness, vision difficulties) on heart disease prevalence and risk factors.
- Explore if difficulty in concentration or physical activities differs based on sensory impairments.

**Smoking and E-Cigarette Usage Patterns:**

- Examine the patterns of smoking and e-cigarette usage and their correlation with heart disease prevalence.
- Are there differences in cardiovascular health among smokers, former smokers, and non-smokers?

**Vaccination and Preventive Measures:**

- Assess the relationship between preventive measures (flu vaccination, pneumococcal vaccination) and heart disease prevalence.
- Explore variations in vaccination rates among different demographic groups.

**Impact of COVID-19:**

- Explore the influence of COVID-19 on cardiovascular health, considering its association with heart disease prevalence and related risk factors.
- Analyze if individuals who tested positive for COVID-19 exhibit distinctive patterns in cardiovascular health.

# Importing Python Libraries

In [6]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

# Loading dataset

In [7]:
# Extract the given zip file if csv file is not present in the dataset folder
heart_data = pd.read_csv('./dataset/heart_2022_with_nans.csv')

# Data Identification and Cleaning

In [10]:
heart_data.head()

State     Sex GeneralHealth  PhysicalHealthDays  MentalHealthDays  \
0  Alabama  Female     Very good                 0.0               0.0   
1  Alabama  Female     Excellent                 0.0               0.0   
2  Alabama  Female     Very good                 2.0               3.0   
3  Alabama  Female     Excellent                 0.0               0.0   
4  Alabama  Female          Fair                 2.0               0.0   

                                     LastCheckupTime PhysicalActivities  \
0  Within past year (anytime less than 12 months ...                 No   
1                                                NaN                 No   
2  Within past year (anytime less than 12 months ...                Yes   
3  Within past year (anytime less than 12 months ...                Yes   
4  Within past year (anytime less than 12 months ...                Yes   

   SleepHours RemovedTeeth HadHeartAttack  ... HeightInMeters  \
0         8.0          NaN             No  ...            NaN   
1         6.0          NaN             No  ...           1.60   
2         5.0          NaN             No  ...           1.57   
3         7.0          NaN             No  ...           1.65   
4         9.0          NaN             No  ...           1.57   

  WeightInKilograms    BMI AlcoholDrinkers HIVTesting FluVaxLast12  \
0               NaN    NaN              No         No          Yes   
1             68.04  26.57              No         No           No   
2             63.50  25.61              No         No           No   
3             63.50  23.30              No         No          Yes   
4             53.98  21.77             Yes         No           No   

  PneumoVaxEver                                  TetanusLast10Tdap  \
0            No  Yes, received tetanus shot but not sure what type   
1            No  No, did not receive any tetanus shot in the pa...   
2            No                                                NaN   
3           Yes  No, did not receive any tetanus shot in the pa...   
4           Yes  No, did not receive any tetanus shot in the pa...   

  HighRiskLastYear CovidPos  
0               No       No  
1               No       No  
2               No      Yes  
3               No       No  
4               No       No  

[5 rows x 40 columns]

In [11]:
heart_data.shape

(445132, 40)

- There are 445132 rows and 40 columns in the dataset.

In [13]:
heart_data.tail()

State     Sex GeneralHealth  PhysicalHealthDays  \
445127  Virgin Islands  Female          Good                 0.0   
445128  Virgin Islands  Female     Excellent                 2.0   
445129  Virgin Islands  Female          Poor                30.0   
445130  Virgin Islands    Male     Very good                 0.0   
445131  Virgin Islands    Male     Very good                 0.0   

        MentalHealthDays                                    LastCheckupTime  \
445127               3.0  Within past 2 years (1 year but less than 2 ye...   
445128               2.0  Within past year (anytime less than 12 months ...   
445129              30.0                                5 or more years ago   
445130               0.0  Within past year (anytime less than 12 months ...   
445131               1.0                                                NaN   

       PhysicalActivities  SleepHours  RemovedTeeth HadHeartAttack  ...  \
445127                Yes         6.0  None of them             No  ...   
445128                Yes         7.0  None of them             No  ...   
445129                 No         5.0        1 to 5             No  ...   
445130                 No         5.0  None of them            Yes  ...   
445131                Yes         5.0  None of them             No  ...   

       HeightInMeters WeightInKilograms    BMI AlcoholDrinkers HIVTesting  \
445127           1.65             69.85  25.63             NaN        Yes   
445128           1.70             83.01  28.66              No        Yes   
445129           1.70             49.90  17.23             NaN         No   
445130           1.83            108.86  32.55              No        Yes   
445131           1.68             63.50  22.60             Yes         No   

       FluVaxLast12 PneumoVaxEver  \
445127           No            No   
445128          Yes            No   
445129           No            No   
445130          Yes           Yes   
445131           No            No   

                                        TetanusLast10Tdap HighRiskLastYear  \
445127  No, did not receive any tetanus shot in the pa...               No   
445128  Yes, received tetanus shot but not sure what type               No   
445129  No, did not receive any tetanus shot in the pa...               No   
445130  No, did not receive any tetanus shot in the pa...               No   
445131  Yes, received tetanus shot but not sure what type               No   

       CovidPos  
445127      Yes  
445128       No  
445129       No  
445130      Yes  
445131       No  

[5 rows x 40 columns]

In [12]:
heart_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 445132 entries, 0 to 445131
Data columns (total 40 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   State                      445132 non-null  object 
 1   Sex                        445132 non-null  object 
 2   GeneralHealth              443934 non-null  object 
 3   PhysicalHealthDays         434205 non-null  float64
 4   MentalHealthDays           436065 non-null  float64
 5   LastCheckupTime            436824 non-null  object 
 6   PhysicalActivities         444039 non-null  object 
 7   SleepHours                 439679 non-null  float64
 8   RemovedTeeth               433772 non-null  object 
 9   HadHeartAttack             442067 non-null  object 
 10  HadAngina                  440727 non-null  object 
 11  HadStroke                  443575 non-null  object 
 12  HadAsthma                  443359 non-null  object 
 13  HadSkinCancer              44

- Most of the columns(variables) are categorical and boolen.
- Physical and mental health days, sleep hours, height, weight, BMI are numeric(float) values.

In [15]:
heart_data.isna().sum()

State                            0
Sex                              0
GeneralHealth                 1198
PhysicalHealthDays           10927
MentalHealthDays              9067
LastCheckupTime               8308
PhysicalActivities            1093
SleepHours                    5453
RemovedTeeth                 11360
HadHeartAttack                3065
HadAngina                     4405
HadStroke                     1557
HadAsthma                     1773
HadSkinCancer                 3143
HadCOPD                       2219
HadDepressiveDisorder         2812
HadKidneyDisease              1926
HadArthritis                  2633
HadDiabetes                   1087
DeafOrHardOfHearing          20647
BlindOrVisionDifficulty      21564
DifficultyConcentrating      24240
DifficultyWalking            24012
DifficultyDressingBathing    23915
DifficultyErrands            25656
SmokerStatus                 35462
ECigaretteUsage              35660
ChestScan                    56046
RaceEthnicityCategor

In [33]:
# Calculating percentage of null values in each column
null_values = round(heart_data.isna().sum()/heart_data.shape[0] * 100,2)

In [34]:
null_values

State                         0.00
Sex                           0.00
GeneralHealth                 0.27
PhysicalHealthDays            2.45
MentalHealthDays              2.04
LastCheckupTime               1.87
PhysicalActivities            0.25
SleepHours                    1.23
RemovedTeeth                  2.55
HadHeartAttack                0.69
HadAngina                     0.99
HadStroke                     0.35
HadAsthma                     0.40
HadSkinCancer                 0.71
HadCOPD                       0.50
HadDepressiveDisorder         0.63
HadKidneyDisease              0.43
HadArthritis                  0.59
HadDiabetes                   0.24
DeafOrHardOfHearing           4.64
BlindOrVisionDifficulty       4.84
DifficultyConcentrating       5.45
DifficultyWalking             5.39
DifficultyDressingBathing     5.37
DifficultyErrands             5.76
SmokerStatus                  7.97
ECigaretteUsage               8.01
ChestScan                    12.59
RaceEthnicityCategor